CMakeLists.txt

```diff
@@ -152,7 +152,7 @@ if (STATICCOMPILE)
 endif()
 add_feature_info(Static STATICCOMPILE "Compiles static library and executable")
 
-if ((${CMAKE_SYSTEM_NAME} MATCHES "Windows") AND (NOT BUILD_SHARED_LIBS))
+if ((${CMAKE_SYSTEM_NAME} MATCHES "Windows") OR (NOT BUILD_SHARED_LIBS))
```

Comment:
I'm not sure about the Windows logic here, but we definitely need to set `Boost_USE_STATIC_LIBS` whenever we are doing a static compilation, even on Linux, or else the build will fail.